lib/mayaUsd/ufe/UsdTransform3d.cpp

```diff
@@ -103,126 +103,89 @@ Ufe::SceneItem::Ptr UsdTransform3d::sceneItem() const
-	{
-		// Initially, attribute can be created, but have no value.
-		GfVec3f v;
-		if (fPrim.GetAttribute(scaleTok).Get<GfVec3f>(&v,getTime(path())))
-		{
-			x = v[0]; y = v[1]; z = v[2];
-		}
-	}
-	return Ufe::Vector3d(x, y, z);
+	return UsdScaleUndoableCommand::create(fItem, GfVec3f(x, y, z));
 }
 
-Ufe::RotateUndoableCommand::Ptr UsdTransform3d::rotateCmd(double x, double y, double z)
+#else // UFE_PREVIEW_VERSION_NUM < 2013
+
+Ufe::TranslateUndoableCommand::Ptr UsdTransform3d::translateCmd()
 {
-	return UsdRotateUndoableCommand::create(fItem, x, y, z);
+	return UsdTranslateUndoableCommand::create(fItem);
 }
-#endif
 
-void UsdTransform3d::rotate(double x, double y, double z)
+Ufe::RotateUndoableCommand::Ptr UsdTransform3d::rotateCmd()
 {
-	rotateOp(fPrim, fItem->path(), x, y, z);
+	return UsdRotateUndoableCommand::create(fItem);
 }
 
-#if UFE_PREVIEW_VERSION_NUM >= 2013
-Ufe::ScaleUndoableCommand::Ptr UsdTransform3d::scaleCmd(double x, double y, double z)
+Ufe::ScaleUndoableCommand::Ptr UsdTransform3d::scaleCmd()
 {
-	return UsdScaleUndoableCommand::create(fItem, x, y, z);
+	return UsdScaleUndoableCommand::create(fItem);
 }
 
-#else
+#endif
 
-Ufe::TranslateUndoableCommand::Ptr UsdTransform3d::translateCmd()
+void UsdTransform3d::translate(double x, double y, double z)
 {
-	return UsdTranslateUndoableCommand::create(fItem, 0, 0, 0);
+	UsdTranslateUndoableCommand(fItem, GfVec3d(x, y, z)).translate(x, y, z);
 }
 
-Ufe::RotateUndoableCommand::Ptr UsdTransform3d::rotateCmd()
+Ufe::Vector3d UsdTransform3d::translation() const
 {
-	return UsdRotateUndoableCommand::create(fItem, 0, 0, 0);
+	return Ufe::Vector3d(0,0,0);
 }
 
-Ufe::ScaleUndoableCommand::Ptr UsdTransform3d::scaleCmd()
+void UsdTransform3d::rotate(double x, double y, double z)
 {
-	return UsdScaleUndoableCommand::create(fItem, 1, 1, 1);
+	UsdRotateUndoableCommand(fItem, GfVec3f(x,y,z)).rotate(x,y,z);
 }
-#endif
 
 void UsdTransform3d::scale(double x, double y, double z)
 {
-	scaleOp(fPrim, fItem->path(), x, y, z);
+	UsdScaleUndoableCommand(fItem, GfVec3f(x,y,z)).scale(x,y,z);
 }
 
 Ufe::TranslateUndoableCommand::Ptr UsdTransform3d::rotatePivotTranslateCmd()
 {
-	auto translateCmd = UsdRotatePivotTranslateUndoableCommand::create(fPrim, fItem->path(), fItem);
-	return translateCmd;
+	return UsdRotatePivotTranslateUndoableCommand::create(fItem);
 }
 
 void UsdTransform3d::rotatePivotTranslate(double x, double y, double z)
 {
-	rotatePivotTranslateOp(fPrim, fItem->path(), x, y, z);
+	UsdRotatePivotTranslateUndoableCommand(fItem, GfVec3f(x,y,z)).translate(x,y,z);
 }
 
 Ufe::Vector3d UsdTransform3d::rotatePivot() const
 {
-	double x{0}, y{0}, z{0};
+	float x{0}, y{0}, z{0};
 	const TfToken xpivot("xformOp:translate:pivot");
 	if (fPrim.HasAttribute(xpivot))
 	{
 		// Initially, attribute can be created, but have no value.
 		GfVec3f v;
 		if (fPrim.GetAttribute(xpivot).Get<GfVec3f>(&v,getTime(path())))
 		{
-			x = v[0]; y = v[1]; z = v[2];
+			std::tie(x, y, z) = { v[0], v[1], v[2] };
```

Comment:
How is this an 'improvement'? 
Keep it simple, Keep it stupid, and assume some poor inturn will have to pick up and maintain this code in the future. Make life easy for them, and write clear, dumb, and maintainable code.

Reply:
Which debugger? (and define 'choke').